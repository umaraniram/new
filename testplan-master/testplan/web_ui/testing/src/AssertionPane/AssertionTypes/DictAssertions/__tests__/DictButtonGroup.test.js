import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import DictButtonGroup from '../DictButtonGroup';

function defaultProps() {
  return {
    uid: 'sort-test-uid',
    defaultSortType: 3,
    sortTypeList: [1, 2, 3, 4],
    flattenedDict: [
      [0,"foo","Passed",["int","1"],["int","1"]],
      [0,"bar","Failed",["int","2"],["int","5"]],
      [0,"extra-key","Failed",[null,"ABSENT"],["int","10"]]
    ],
    setRowData: jest.fn()
  };
}

describe('DictLogAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = {};
    shallowComponent = undefined;
  });

  it('shallow renders DictButtonGroup component', () => {
    props = defaultProps();
    shallowComponent = shallow(<DictButtonGroup {...props} />);
    expect(shallowComponent).toMatchSnapshot();
  });
});