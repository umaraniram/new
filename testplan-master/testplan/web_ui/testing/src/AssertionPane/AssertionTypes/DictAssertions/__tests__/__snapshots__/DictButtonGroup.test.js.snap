// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictLogAssertion shallow renders DictButtonGroup component 1`] = `
<ButtonGroup
  role="group"
  style={
    Object {
      "paddingBottom": ".5rem",
    }
  }
  tag="div"
>
  <Button
    active={false}
    color="secondary"
    key="sort-test-uid1"
    onClick={[Function]}
    outline={true}
    size="sm"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="sort-amount-down"
      inverse={false}
      key="faSortAmountDown"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="sm"
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Button>
  <Button
    active={false}
    color="secondary"
    key="sort-test-uid2"
    onClick={[Function]}
    outline={true}
    size="sm"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="sort-amount-up"
      inverse={false}
      key="faSortAmountUp"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="sm"
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Button>
  <Button
    active={true}
    color="secondary"
    key="sort-test-uid3"
    onClick={[Function]}
    outline={true}
    size="sm"
    tag="button"
  >
    Status
  </Button>
  <Button
    active={false}
    color="secondary"
    key="sort-test-uid4"
    onClick={[Function]}
    outline={true}
    size="sm"
    tag="button"
  >
    Failures only
  </Button>
  <CopyButton
    value="<table><tr><th>Key</th><th>Expected</th><th>Value</th></tr><tr><td></td><td><small></small></td><td><small></small></td></tr><tr style=\\"color: red;\\"><td></td><td><small></small></td><td><small></small></td></tr><tr style=\\"color: red;\\"><td></td><td><small></small></td><td><small></small></td></tr></table>"
  />
</ButtonGroup>
`;
